DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+// Name registrar using deposits
+
+contract Deed {
+    // The Deed is a contract intended simply to hold ether
+    // It can be controlled only by the registrar and can only send ether back to the owner
+    address public owner;
+    address public bidder;
+    Registrar public registrar;
+    bytes32 public sealedBid;
+    bytes32 public hashedName;
+    uint public value;
+    address public burn = 0xdead;
+    uint public creationDate;
+    
+    modifier noEther {
+        if (msg.value > 0) throw;
+        _
+    } 
+    
+    modifier onlyRegistrar {
+        if (msg.sender != address(registrar)) throw;
+        _
+    }
+    
+    function Deed(address _bidder, bytes32 _sealedBid, address _registrar) {
+        bidder = _bidder;
+        registrar = Registrar(_registrar);
+        sealedBid = _sealedBid;
+        creationDate = now;
+    }
+    
+    function unsealBid(bytes32 _hashedName, address _owner, uint _value, bytes32 _salt) noEther onlyRegistrar {
+        if (sha3(_hashedName, _owner, _value, _salt) != sealedBid) throw;
+        hashedName = _hashedName;
+        owner = _owner;
+        value = _value;
+        if (!bidder.send(this.balance - _value)) throw;
+    }
+    
+    function closeDeed( uint burnAmount) noEther onlyRegistrar {
+        burn.send((burnAmount * this.balance)/1000);
+        selfdestruct(bidder);
```

Comment:
mmm didn't know that! I can reverse the process, I send funds to the owner and then burn by selfdetruct.

I wanted to do this way to avoid any funny business with using a fallback function, but we can try to be extra careful;